yt/data_objects/tests/test_chunking.py

```diff
@@ -38,3 +40,20 @@ def test_chunking():
             assert_equal(coords['f']['io'], coords['f']['spatial'])
             assert_equal(coords['i']['io'], coords['i']['all'])
             assert_equal(coords['i']['io'], coords['i']['spatial'])
+
+def test_ds_hold():
+    ds1 = fake_random_ds(64)
+    ds2 = fake_random_ds(128)
+    ds1wr = weakref.proxy(ds1)
+    ds2wr = weakref.proxy(ds2)
+    dd = ds1.all_data()
+    assert_true(dd.ds == ds1wr) # dd.ds is a weakref, so can't use "is"
```

Comment:
Maybe instead of trying to compare weakrefs which seems to be problematic on windows and on py2 you could look at ds.domain_dimensions or something?

Reply:
That would work in this particular case.  I'll have a shot at it.